### PR TITLE
Add memory swap param

### LINKED ISSUE #410: set the "memory-swap" parameter in docker containers.
Falling out of internal ticket PAASTA-3595.

By default, Docker sets the the value of ``memory.memsw.limit_in_bytes`` to the memory limit *2. This is undocumented, but can be seen [here](https://github.com/docker/docker/blob/master/daemon/daemon_unix.go#L292-L294).

We should fix this, ensuring that the memory limit set in soa-configs is memory+swap, rather than just memory so that containers are killed when they exceed that memory limit rather than swapping. We can do this by setting the ``--memory-swap`` parameter passed to the ``docker run`` invocation.

To do this in marathon and chronos, we can use the ``parameters`` argument available to pass to containers:

- [marathon](https://mesosphere.github.io/marathon/docs/native-docker.html#privileged-mode-and-arbitrary-docker-options)
- [chronos - undocumented, but in the code](https://github.com/mesos/chronos/blob/fc2567dbfb0567fd35a496be09a39474c3d0a7c6/src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Containers.scala#L30)

Unfortunately, this will cause a big bounce of applications.

### PR DESCRIPTION
**Pull request by Rob-Johnson**
closes #410 

I'm not sure the itests tell me much because of #373 ? I'm testing on mesosstage.

### CODE REVIEW

File: paasta_tools/chronos_tools.py [hunk 1]
```diff
@@ -392,7 +392,10 @@ def format_chronos_job_dict(self, docker_url, docker_volumes, dockerfile_locatio
                 'image': docker_url,
                 'network': net,
                 'type': 'DOCKER',
-                'volumes': docker_volumes
+                'volumes': docker_volumes,
+                'parameters': {
+                    'memory-swap': "%sm" % self.get_mem(),
```
Comment (solarkennedy) [inline paasta_tools/chronos_tools.py]: The docker man page says this "always" must be larger than -m, can you confirm it actually works if this is the same?


Also can you update the docs about "how paasta runs docker images to say that paasta doesn't allow swap?

**Comment by solarkennedy** [comment]
I'm not afraid of the big bounce, just ship it slowly a couple clusters at a time like normal.

**Comment by solarkennedy** [comment]
ship

File: paasta_tools/chronos_tools.py (hunk 1 above)
Comment (Rob-Johnson) [inline paasta_tools/chronos_tools.py]: confirmed launching a marathon task - it works fine.

**Comment by Rob-Johnson** [comment]
Chronos doesn't seem to do the right thing here:

here's the test job that was posted (Ive removed the volume mounts)

```
(py27)bash-4.1# ./paasta_tools/setup_chronos_job.py example_service.mesosstage_robj_test
Job to update: {'retries': 2, 'shell': True, 'container': {'image': u'docker-paasta.yelpcorp.com:443/services-example_service:paasta-4bacf539ce5184bd9fa1862db5afb9aad50703cc', 'parameters': {'memory-swap': '100m'}, 'type': 'DOCKER', 'network': 'BRIDGE', 'volumes': []}, 'name': 'example_service mesosstage_robj_test', 'schedule': 'R/2015-08-14T20:35:04+00:00/PT1M', 'mem': 100, 'epsilon': 'PT60S', 'scheduleTimeZone': None, 'description': 'config5812fbc7', 'cpus': 0.1, 'environmentVariables': [{'name': 'PAASTA_INSTANCE', 'value': 'mesosstage_robj_test'}, {'name': 'PAASTA_SERVICE', 'value': 'example_service'}, {'name': 'PAASTA_CLUSTER', 'value': u'mesosstage'}], 'disabled': False, 'command': 'sleep 2m', 'arguments': None, 'owner': 'noop', 'async': False, 'disk': 1024, 'uris': ['file:///root/.dockercfg'], 'constraints': [['pool', 'EQUALS', 'default']]}
Updated Chronos job: example_service mesosstage_robj_test
````

Note the ``'parameters': {'memory-swap': '100m'}``.

But the parameters never make it to the executor from what I can tell (the docker container doesn't have the memory-swap param set as it should) and querying the job via the API shows Chronos knows nothing of the params:

```
$ curl  -H 'Accept: application/json' 'http://chronos.paasta-mesosstage.yelp/scheduler/jobs'
```
screen shot 2016-04-12 at 17 08 24 1

File: paasta_tools/chronos_tools.py (hunk 1 above)
Comment (Rob-Johnson) [inline paasta_tools/chronos_tools.py]: also confirmed this works when the kernel cmdline option isn't set

```

robj@srv5-r6-sfo2:~  % cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-3.13.0-65-generic root=UUID=6e4297c5-c9a3-4bb3-bddc-0ecec13a2349 ro console=ttyS0,115200 console=tty0 elevator=noop
robj@srv5-r6-sfo2:~  % sudo docker run -it --memory 100m --memory-swap 100m ubuntu bash
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
87192bdbe00f: Already exists
28e09fddaacb: Already exists
7e15ce58ccb2: Already exists
a3ed95caeb02: Already exists
Digest: sha256:f6e8757419147ba099af8cec4db365b3603472bd6182722b16fdae932c0bf3bf
Status: Downloaded newer image for ubuntu:latest
WARNING: Your kernel does not support swap limit capabilities, memory limited without swap.
```

**Pull request merged**